### PR TITLE
[issue-150] Add Hadoop TeraSort support using Pravega Streams

### PR DESCRIPTION
**Pull request by kevinhan88**
[issue-150] Add Hadoop TeraSort support using Pravega Streams

### CODE REVIEW

File: hadoop-connector-examples/README.md [hunk 1]
```diff
@@ -101,3 +101,38 @@ You can also use hadoop-connectors on Spark
 Spark (verified with Spark 2.2.1 on Ubuntu 16.04)
    spark-submit --class io.pravega.examples.spark.WordCount ${HADOOP_EXAMPLES_JAR} ${PRAVEGA_URI} ${PRAVEGA_SCOPE} ${PRAVEGA_STREAM}
 ```
```
Comment (RaulGracia) [inline hadoop-connector-examples/README.md]: Please, update the `Hadoop Connector Examples` table in main `README` by adding this new sample.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Wed Mar 15 15:48:47 PDT 2017
```
Comment (RaulGracia) [inline gradle/wrapper/gradle-wrapper.properties]: Why this file has been updated?

File: hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/terasort/TeraGen.java [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.example.hadoop.terasort;
+
+import io.pravega.example.hadoop.wordcount.PravegaOutputFormat;
+import io.pravega.example.hadoop.wordcount.TextSerializer;
```
Comment (RaulGracia) [inline hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/terasort/TeraGen.java]: If `PravegaOutputFormat` and `TextSerializer` are common utilities, maybe they should be in another package, so any new hadoop example may easily use them.

File: hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/wordcount/ExampleDriver.java [hunk 4]
```diff
@@ -38,6 +41,10 @@ public static void main(String argv[]) {
                     "A map/reduce program that writes random data to pravega.");
             pgd.addClass("randomtextwriter", RandomTextWriter.class,
                     "A map/reduce program that writes random textual data to pravega.");
+            pgd.addClass("teragen", TeraGen.class,
+                    "A map/reduce program that generate the official GraySort input data set to pravega.");
+            pgd.addClass("terastreamvalidate", TeraStreamValidate.class,
```
Comment (RaulGracia) [inline hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/wordcount/ExampleDriver.java]: May `terasort` be missing here?

File: hadoop-connector-examples/README.md (hunk 1 above)
Comment (kevinhan88) [reply hadoop-connector-examples/README.md]: Added in the following commit

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (kevinhan88) [reply gradle/wrapper/gradle-wrapper.properties]: Changed it back

File: hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/terasort/TeraGen.java (hunk 3 above)
Comment (kevinhan88) [reply hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/terasort/TeraGen.java]: Yes there is a backlog item for it.
For the moment for the sake of terasort test, let's just leave the change here.

File: hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/wordcount/ExampleDriver.java (hunk 4 above)
Comment (kevinhan88) [reply hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/wordcount/ExampleDriver.java]: Added

**Comment by kevinhan88** [comment]
Just close this pull request due to missing DCO, and then create a new one

**Pull request closed without merging**